packages/xchain-radix/src/client.ts

```diff
@@ -0,0 +1,482 @@
+import { CommittedTransactionInfo, GatewayApiClient } from '@radixdlt/babylon-gateway-api-sdk'
+import {
+  LTSRadixEngineToolkit,
+  NetworkId,
+  PrivateKey,
+  RadixEngineToolkit,
+  SimpleTransactionBuilder,
+} from '@radixdlt/radix-engine-toolkit'
+import {
+  AssetInfo,
+  Balance,
+  BaseXChainClient,
+  FeeOption,
+  FeeType,
+  Fees,
+  Network,
+  PreparedTx,
+  Tx,
+  TxHistoryParams,
+  TxParams,
+  TxType,
+  TxsPage,
+  XChainClientParams,
+} from '@xchainjs/xchain-client'
+import { getSeed } from '@xchainjs/xchain-crypto/lib'
+import { Address, Asset, assetAmount, assetToBase, baseAmount } from '@xchainjs/xchain-util'
+import axios from 'axios'
+import {
+  AssetXRD,
+  KeyType,
+  MAINNET_GATEWAY_URL,
+  RadixChain,
+  STATE_ENTITY_DETAILS_PATH,
+  STOKENET_GATEWAY_URL,
+  TRANSACTION_COMMITTED_DETAILS_PATH,
+  TRANSACTION_CONSTRUCTION_PATH,
+  TRANSACTION_PREVIEW_PATH,
+  TRANSACTION_SUBMIT_PATH,
+  TransferTransactionManifest,
+  XRD_DECIMAL,
+} from './const'
+import { EntityDetailsResponse, Transaction } from './types/radix'
+
+/**
+ * Custom Radix client
+ */
+
+class Client extends BaseXChainClient {
+  privateKey: PrivateKey
+  gatewayApiClient: GatewayApiClient
+  constructor(params: XChainClientParams, keyType: KeyType) {
+    super(RadixChain, { network: params.network, phrase: params.phrase })
+    const seed = getSeed(this.phrase)
+    const hexString = seed.toString('hex').substring(0, 64)
+    if (keyType == KeyType.Ed25519) {
+      this.privateKey = new PrivateKey.Ed25519(hexString)
+    } else {
+      this.privateKey = new PrivateKey.Secp256k1(hexString)
+    }
+    this.gatewayApiClient = GatewayApiClient.initialize({
+      networkId: this.getRadixNetwork(),
+      applicationName: 'xchainjs',
+    })
+  }
+  /**
+   * Get an estimated fee for a test transaction that involves sending
+   * XRD from one account to another
+   *
+   * @returns {Fee} An estimated fee
+   */
+  async getFees(): Promise<Fees> {
+    const transactionPreviewUrl = `${this.getGatewayUrl()}${TRANSACTION_PREVIEW_PATH}`
```

Comment:
Hmm... we have a gateway client in the client's state but we do not seem to be using it.